### DISCUSSION TITLE
Running CFAST through a python interface

### DISCUSSION THREAD

**Discussion by Sari-Alkhatib** [firemodels/cfast#2119] [Q&A]
Hi,

I am working on a sensitivity analysis using CFAST. I am using python to run the whole analysis which runs CFAST for various input files. Is there any recommendation to manage the runs in terms of terminating specific runs using python commands? For example, restricting the runs to a specific upper time limit. Any feedback or recommendation would be helpful. 

Thanks!

**Comment by preneke** [comment]
CFAST comes with a program CData that is meant to run Monte Carlo simulations. I understand if you already have a program set up. You can include in the directory you are running the case a stop file, <case name>.stop with a single line that has the maximum number of steps you want the program to run. I would suggest something large like half a million. Sometimes, not often, the code can speed up again and finish.

**Reply by Sari-Alkhatib** [reply]
Thanks! 
I am aware about CData, but I was not aware about the .stop feature in it. The decision on not using CData was mainly due to the parameters we are interested in and how they are sampled. 
You recommend half a million "time steps"? (for 1 hr long compartment fires)

**Reply by rpeacoc** [reply]
The .stop is actually a feature in CFAST. It should work outside of CData too.

**Reply by Sari-Alkhatib** [reply]
The .stop feature sounds interesting. I am not sure if I read about it in either CFAST or CData docs. I think I am currently using a python termination command based on the wall-clock (which I guess is a similar approach).

**Comment by preneke** [comment]
Sorry, somehow part of my response was deleted. If your case is named A.in then the stop file should be A.stop and not just .stop as the name. Sorry for the missing text.

**Comment by preneke** [comment]
Half a million is a bit arbitrary. I suspect you probably need to do some experimentation. I should have said, instead of recommending a value, start big and work down. My perspective is that it is better to take a little longer than to drop cases you could have included. 

What variables and distributions are you using?

**Reply by Sari-Alkhatib** [reply]
I think my understanding about your initial estimation (as an expert opinion) was correct. So, thanks for the clarification!!

I think the whole sampling approach I wanted did not match my understanding of CData approach. The two main examples are 1) the compartments configuration, e.g., how they are aligned and how to control this alignment as I am centering them and a connecting vent on one central point, and 2) the sampling of the fire HRR time histories/profiles (I am using a specific data to define four points of the HRR profile that I am sampling, namely, the point at which the fire can be considered developed, maximum, the point at which the fire starts to decay, and the final value of the HRR which will stay until the end of the 1 hr).  

So far, I am using uniform distributions but with some biased sampling.

The context of this study is very specific so I decided to do my own code (it is strongly connected to the ventilation conditions).
